cmake/taget_info.cmake

```diff
@@ -0,0 +1,389 @@
+function(echo_target_property tgt prop)
+    # v for value, d for defined, s for set
+    get_property(v TARGET ${tgt} PROPERTY ${prop})
+    get_property(d TARGET ${tgt} PROPERTY ${prop} DEFINED)
+    get_property(s TARGET ${tgt} PROPERTY ${prop} SET)
+    # only produce output for values that are set
+    if(s)
+        message("Property = '${prop}' Defined='${d}' Set='${s}'")
+        message("   Value = '${v}'")
+    endif()
+endfunction()
+
+function(echo_target tgt)
+    set(props
```

Comment:
Is this list of properties taken from somewhere? Could be good to add a comment in that case for possible future modifications